src/explore-education-statistics-common/src/modules/table-tool/contexts/TableHeadersContext.tsx

```diff
@@ -0,0 +1,109 @@
+import useToggle from '@common/hooks/useToggle';
+import noop from 'lodash/noop';
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+
+export interface TableHeadersContextState {
+  activeList?: string;
+  defaultNumberOfItems: number;
+  expandedLists: string[];
+  groupDraggingActive: boolean;
+  groupDraggingEnabled: boolean;
+  setActiveList: (listName?: string) => void;
+  toggleExpandedList: (listName: string) => void;
+  toggleGroupDraggingActive: () => void;
+  toggleGroupDraggingEnabled: () => void;
```

Comment:
Same as above - would suggest requiring a boolean parameter like `enabled` to determine the next state.